### ISSUE TITLE
How to use after installation?

### ISSUE THREAD

**Issue by rlam3** [steelbrain/autocomplete-swift#5]
How to use after installation? I'm not able to see any autocompletion. What other steps need to be taken in order for auto complete to work? Thanks!

**Comment by steelbrain** [comment]
Have you installed [SourceKittenDaemon](https://github.com/terhechte/SourceKittenDaemon)?

**Comment by rlam3** [comment]
Cool. Does this work with Cocoapods too?

**Comment by steelbrain** [comment]
Not really sure, I've never really used this package. I only wrote this because someone requested this in the Atom Slack and agreed to test it after I wrote it 😉 
In any case, the dependency I mentioned above is required to use this package

**Comment by sahil-a** [comment]
I installed SourceKittenDaemon but i cant figure out how to get it to work?

**Comment by steelbrain** [comment]
@sahil-a Do you have `.xcodeproj` in your project root?

**Comment by sahil-a** [comment]
No, I was using SPM.

**Comment by steelbrain** [comment]
Well, that is a requirement for both SourceKittenDaemon and this package

**Comment by sahil-a** [comment]
Oh, ok. Thanks! I will try that out.

**Comment by jnewc** [comment]
Just run into this problem too. Requiring an xcodeproj seems quite limiting - 99% of projects with an xcodeproj will be developed in Xcode, whereas SPM build setups and scripts are more likely to be developed in Atom.

Could this restriction be lifted or is the xcodeproj file critical to the implementation?

**Comment by steelbrain** [comment]
@jnewc This package is just a proxy between Atom and [`SourceKittenDaemon`](https://github.com/terhechte/SourceKittenDaemon). If you want this package to work with your project without xcodeproj, Please open up a request in that repo and ask for it

**Comment by jnewc** [comment]
Thanks @steelbrain, will do

**Comment by sahil-a** [comment]
I tried this again out of curiosity. I had to change the port to 8080 and run the SourceKittenDaemon, and I saw that it was getting GET requests, but I wasn't getting autocompletion. Has this been tested?

**Comment by steelbrain** [comment]
@sahil-a Yes this has been tested well

**Comment by monomadic** [comment]
I have to kind of push the sentiment of this post. If you're going to write a package, write some basic documentation on how to install the thing or don't bother. I've installed SKD (which I worked out I had to do because of getting error messages upon installation) and rebooted Atom only to be completely confused what to do next.

When I try to run SKD from the terminal, I get this:
```
dyld: Library not loaded: @rpath/libswiftCore.dylib
  Referenced from: /usr/local/Cellar/sourcekitten/0.16.0/Frameworks/SourceKittenFramework.framework/Versions/A/SourceKittenFramework
  Reason: Incompatible library version: SourceKittenFramework requires version 1.0.0 or later, but libswiftCore.dylib provides version 0.0.0
```

Very helpful error message - version 0.0.0? what? I realise now I've strayed into this being SKD's issue, but a few paragraphs of documentation would be highly appreciated, Atom is buggy and frustrating enough as it is :)

**Comment by steelbrain** [comment]
@robsaunders Thank you for your kind words, they really appreciate me and make me want to spend more of my time on this project that I work on for free

As it is evident from the error message you've posted above, the error is not in Atom, this package or something that I personally wrote. It's an error about libraries not being loaded, which should be opened directly on SourceKittenDaemon repo as this specific package is just a proxy to that.

I don't mean to be rude, it's just that it feels unpleasant when someone implies everything that's happening to them is my fault. I love helping people, I would love to help you diagnose your issue as well but blaming doesn't fix your problems.

**Comment by monomadic** [comment]
Hey, I and others really appreciate that this plugin exists at all, I didn't mean to convey any rudeness, and certainly not blame, honestly. But it seems it's a common attitude with open source - it's free so don't put effort into polish / coherence, and if someone criticises or suggests improvement in anything but deep careful reverence then it's met with offence. But I do apologise for being terse.

Actually none of this is about 'blame' it's about improving your project for others. As it stands this is the only way to have autocomplete for swift in atom, so is there some kind of acceptance of responsibility for the package maintainers to at least be open to adding documentation? I think so, maybe you differ.

My point wasn't where the error occurred, just that many users who use plugins like this probably just need a few sentences to kick them in the right direction to find out. I had to do some irrational things to actually find out what was wrong, and nowhere did it state the projects dependencies (I had sourcekitten installed for example, but not sourcekittendaemon, not realising they are separate projects).

A readme is just as much a part of the project as anything else and simple documentation really saves people time from messing around. It isn't like I could contribute by writing one myself as I didn't even get to the stage where I knew what to write - I'd love to actually but I don't feel even slightly educated about how the project works, so posting in here is the best I can offer. Maybe someone else will read it and get further than me, because they read this while looking for a readme.

You've spent so much time creating the plugin itself, and really - thank you. I didn't get a chance to try it because I gave up eventually on swift in atom. But why not take 3-5 minutes to save others potentially hours of poking around? As an open source, public project, it is just as much a part of the codebase (and in some ways more important) than the code itself.
